# Eliminacja wartości odstających

means =

 Columns 1 through 6:

   1.8679e-01   1.4839e-02   2.1045e-01   2.0882e-01   7.9658e+01   1.0604e+00

 Column 7:

   9.0846e-03

ans = min 0.125 for class 1 is at 642
ans = min 0 for class 2 is at 642
ans = min 0 for class 3 is at 642
ans = min 0 for class 4 is at 642
ans = min -5.4538e-09 for class 5 is at 393
ans = min -1.214e-07 for class 6 is at 25
ans = min -3.26424e-09 for class 7 is at 539
ans = max 5.12127 for class 1 is at 186
ans = max 25.8229 for class 2 is at 186
ans = max 382.139 for class 3 is at 186
ans = max 380.858 for class 4 is at 186
ans = max 145297 for class 5 is at 186
ans = max 1934.12 for class 6 is at 186
ans = max 16.5702 for class 7 is at 186
means =

 Columns 1 through 6:

   1.8404e-01   6.9109e-04   9.4984e-04   1.5044e-05  -7.5376e-10   2.9695e-07

 Column 7:

  -9.6408e-11

ans = min 0.148148 for class 1 is at 186min 642 for class 
ans = min 3.51422e-09 for class 2 is at 704
ans = min 7.53088e-11 for class 3 is at 816
ans = min 5.13337e-11 for

Usuwam ze zbioru uczącego próbki 642 (wartości skrajnie małe dla większości klas) oraz 186 (wartości skrajnie duże dla większości klas). Sensowność usunięcia tych dwóch próbek widoczna jest skali wykresów generowanych przez plot2features. Przed eliminacją dane tworzyły dwa zgrupowania – punktów odstających i reszty. Po usunięciu wartości odstających skala wykresów pozwala zauważyć to jak wyglądają różnice między klasami.


In [ ]:
for i=1:size(Combs)
    ttemp = train;
    testtemp = test;

    first_idx = Combs(i, 1);
    second_idx = Combs(i, 2);

    ttemp = ttemp(:, [1 first_idx second_idx]);
    testtemp = testtemp(:, [1 first_idx second_idx]);

    pdfindep_para = para_indep(ttemp);
    pdfmulti_para = para_multi(ttemp);
    pdfparzen_para = para_parzen(ttemp, 0.001);


    % wyniki do punktu 3
    base_ercf = zeros(1,3);
    base_ercf(1) = mean(bayescls(testtemp(:,2:end), @pdf_indep, pdfindep_para) != testtemp(:,1));
    base_ercf(2) = mean(bayescls(testtemp(:,2:end), @pdf_multi, pdfmulti_para) != testtemp(:,1));
    base_ercf(3) = mean(bayescls(testtemp(:,2:end), @pdf_parzen, pdfparzen_para) != testtemp(:,1));
    disp(Combs(i, :))
    disp(base_ercf)
end

  # Błędy dla kombinacji
   
   |kombinacja | pdfindep_para | pdfmulti_para | pdfparzen_para|
   |-----------|---------------|---------------|---------------|
   |2   3      |0.17818        |0.17873        |   0.21765     |
   |2   4      |0.0263158      |0.0049342      |0.0241228      |
   |2   5      |0.075658       |0.059759       |0.270833       |
   |2   6      |0.13706        |0.10197        |0.27686        |
   |2   7      |0.102522       |0.097588       |0.276864       |
   |2   8      |0.22752        |0.22862        |0.27686        |
   |3   4      |0.021382       |0.020833       |0.016996       |
   |3   5      |0.13158        |0.12774        |0.23794        |
   |3   6      |0.15570        |0.14912        |0.31469        |
   |3   7      |0.16228        |0.15406        |0.31469        |
   |3   8      |0.28125        |0.28399        |0.31469        |
   |4   5      |0.15406        |0.14583        |0.23958        |
   |4   6      |0.15844        |0.15132        |0.26864        |
   |4   7      |0.23136        |0.21217        |0.26864        |
   |4   8      |0.21272        |0.21272        |0.26864        |
   |5   6      |0.37500        |0.31305        |0.47149        |
   |5   7      |0.32511        |0.26919        |0.47149        |
   |5   8      |0.38158        |0.37829        |0.47149        |
   |6   7      |0.32675        |0.28947        |0.54331        |
   |6   8      |0.40077        |0.40241        |0.57346        |
   |7   8      |0.36842        |0.36678        |0.54331        |
   
Najbardziej korzystna jest zatem kobinacja cech 3 i 4.

Dla apriori = 0.25

  |kombinacja | pdfindep_para | pdfmulti_para | pdfparzen_para|
  |-----------|---------------|---------------|---------------|
  | 3  4      | 0.021382      | 0.020833      |0.016996       |
  | 4  6 | 0.158443  | 0.151316 |0.268640| 
  | 3  4 |0.021382|0.020833|0.016996|

# Klasyfikacja w oparciu o dwie wybrane cechy

Okno Parzena 0.001

|cechy|pdfindep|pdfmulti|pdfparzen|
|---|---|---|---|
|3  4|0.021382|0.020833|0.016996|

używając etykiet klienta

|cechy|pdfindep|pdfmulti|pdfparzen|
|---|---|---|---|
|3  4|0.010417|0.009868|0.006031|

# Redukcja zbioru trenującego

Uśrednione wyniki dla klasyfikacji z użyciem zredukowanego zbioru trenującego dla pięciu prób 

|czesc|pdfindep|std|min|max|
|---|---|----|---|---|
|0.1000|0.0257|0.0049|0.0203|0.0323|
|0.2500|0.0205|0.0011|0.0192|0.0219|
|0.5000|0.0213|0.0008|0.0203|0.0225|

|czesc|pdfmulti|std|min|max|
|---|---|----|---|---|
|0.1000|0.0258|0.0046|0.0214|0.0329|
|0.2500|0.0202|0.0009|0.0192|0.0214|
|0.5000|0.0197|0.0013|0.0181|0.0214|


Okno Parzena 0.001

|czesc|pdfparzen|std|min|max|
|---|---|----|---|---|
|0.1000|0.0402|0.0044|0.0351|0.0471|
|0.2500|0.0283|0.0022|0.0252|0.0312|
|0.5000|0.0203|0.0008|0.0197|0.0214|

używając etykiet klienta

|czesc|pdfindep|std|min|max|
|---|---|----|---|---|
|0.1000|0.0150|0.0057|0.0088|0.0241|
|0.2500|0.0102|0.0010|0.0088|0.0115|
|0.5000|0.0103|0.0010|0.0088|0.0110|


|czesc|pdfmulti|std|min|max|
|---|---|----|---|---|
|0.1000|0.0135|0.0053|0.0082|0.0225|
|0.2500|0.0089|0.0012|0.0071|0.0099|
|0.5000|0.0098|0.0011|0.0088|0.0115|

Okno Parzena 0.001

|czesc|pdfparzen|std|min|max|
|---|---|----|---|---|
|0.1000|0.0249|0.0020|0.0219|0.0214|
|0.2500|0.0167|0.0024|0.0137|0.0214|
|0.5000|0.0091|0.0012|0.0071|0.0214|

# Różne szerokości okna Parzena

|parzen width|error|
|------|-------|
|0.000100|0.014254|
|0.000500|0.016996|
|0.001000|0.020285|
|0.005000|0.044408|
|0.010000|0.043311|

używając etykiet klienta

|parzen width|error|
|-------|------|
|0.000100|0.004386|
|0.000500|0.005482|
|0.001000|0.008772|
|0.005000|0.031250|
|0.010000|0.030154|

# Dwukrotnie większe prawdopodobieństwo apriori dla maści czarnych

Uśrednione dla pięciu prób wyniki klasyfikacji uzyskane przy użyciu zredukownego zbioru trenującego 

|pdfindep|std|min|max|
|---|---|---|----|
|0.0156|0.0034|0.0102|0.0183|

|pdfmulti|std|min|max|
|---|---|---|---|
|0.0167|0.0024|0.0132|0.0190|


Okno Parzena 0.001

|pdfparzen|std|min|max|
|---|---|---|---|
|0.0186|0.0032|0.0139|0.0227|

etykiety klienta:

|pdfindep|std|min|max|
|---|---|---|----|
|0.0076|0.0020|0.0044|0.0095|

|pdfmulti|std|min|max|
|---|---|---|---|
|0.0082|0.0015|0.0066|0.0095|


Okno Parzena 0.001

|pdfparzen|std|min|max|
|---|---|---|---|
|0.0107|0.0010|0.0095|0.0117|

# Normalizacja danych

Odchylenie standardowe dla cech 3 i 4: 9.20618518943410E-04   9.51290337161051E-04

Odchylenie standardowe w poszczególnych klasach dla 

|klasa|std cecha 3|std cecha 4|
|---|---|---|
|1|0.000063|0.000186|
|2|0.000162|0.000003|
|3|0.000022|0.000085|
|4|0.000012|0.000098|
|5|0.000023|0.000021|
|6|0.000255|0.000002|
|7|0.000010|0.000112|
|8|0.000126|0.000009|

błędy klasyfikacji dla znormalizowanych cech 3 i 4

Okno Parzena 0.001

|cechy|pdfindep|pdfmulti|pdfparzen|
|----|---|---|----|
|3  4|0.021382|0.020833|0.020285|

błędy klasyfikacji dla znormalizowanych cech 3 i 4 korzystając z etykiet klienta


Okno Parzena 0.001

|cechy|pdfindep|pdfmulti|pdfparzen|
|----|---|---|---|
|3  4|0.010417|0.009868|0.004386|



